src/datasets/builder.py

```diff
@@ -122,13 +122,16 @@ class BuilderConfig:
             Path(s) to source data file(s).
         description (`str`, *optional*):
             A human description of the configuration.
+        return_file_name (`bool`, defaults to `False`):
+            Whether to return the data file name as a feature.
```

Comment:
Cool ! Maybe add this argument only to the subclasses CsvConfig, ParquetConfig, etc. ?

This way we keep this class generic

Reply:
I moved the definition to the subclasses however, I did not put comments for the new attribute to keep consistency with other attributes.